adafruit_bno055.py

```diff
@@ -196,6 +197,22 @@ def mode(self):
         """
         return self._read_register(_MODE_REGISTER)
 
+    @property
+    def calibration_status(self):
+        """Tuple containing sys, gyro, accel, and mag calibration data."""
+        calibration_data = self._read_register(_CALIBRATION_REGISTER)
+        sys = (calibration_data >> 6) & 0x03
+        gyro = (calibration_data >> 4) & 0x03
+        accel = (calibration_data >> 2) & 0x03
+        mag = calibration_data & 0x03
+        return sys, gyro, accel, mag
```

Comment:
Depends on level of information you want exposed to user. Data sheet doesn’t seem to specify if 0 - 3 are valid values for the calibration status, just that 0 means not calibrated and 3 means fully calibrated. But, we could not do the bitwise and with 0x03 and then just return the actual register values after bitshifting to account for each values offset. That way they could see if certain facets of the calibration are failing and if so, by how much. This is all assuming 0, 1, 2, and 3 are valid as info contained in the calibration register.

Reply:
Data sheet is here https://cdn-learn.adafruit.com/assets/assets/000/036/832/original/BST_BNO055_DS000_14.pdf

Reply:
The datasheets seems to imply that it's either 0 or 3. Why the extra bits? I don't know. Could be for future use or a re-used register layout from a variant that has more status. Have you seen anything that would indicate 1 or 2 are possible and what they mean for the BNO055?

Reply:
No I haven’t seen anything to suggest they’re valid, but they’re not called out as invalid either. The only thing that made me consider it is that the C library for this sensor checks for < 3 instead of == 3. Wonder if we could verify with hardware, or if it’s worth it for the use case for this code.

Reply:
https://github.com/adafruit/Adafruit_BNO055/blob/master/Adafruit_BNO055.cpp#L611

C lib